### PR TITLE
AEA-374 Code Improvements

### PR DESCRIPTION
**Pull request by booshibooshi**
and publish the collection in github release

## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,8 +5,10 @@
 **/public/
 **/coverage/
 **/node_modules/
+models/examples/**/PrepareRequest-FhirMessageUnsigned.json
 tests/e2e/pact/pact/
 tests/e2e/postman/electronic-prescription-coordinator-postman-tests.json
+tests/e2e/postman/collections/electronic-prescription-service-collection.json
```
Comment (booshibooshi) [inline .gitignore:11]: For potential future use, deciding where to make public the postman collection for a given build. Possibly github release

File: Makefile [hunk 2]
```diff
@@ -18,14 +18,16 @@ release:
 	mkdir -p dist
 	cp -r specification/dist/. dist
 	cp -r terraform dist
+	rsync -av --progress --copy-links tests/e2e/pact dist --exclude node_modules
```
Comment (booshibooshi) [inline Makefile:21]: Moving pact publish and verify into release only, to allow for uuid generation at deploy time

File: Makefile [hunk 3]
```diff
@@ -52,20 +54,19 @@ install-hooks:
...
-	$(foreach directory, $(wildcard models/examples/*), cp -r $(directory) models/dist/examples;)
-	$(foreach file, $(wildcard models/schemas/*.yaml), cp $(file) models/dist/schemas;)
-	$(foreach file, $(wildcard models/schemas/*.json), cp $(file) models/dist/schemas;)
-	$(foreach file, $(wildcard models/schemas/*.yaml), poetry run python scripts/yaml2json.py $(file) models/dist/schemas;)
+	$(foreach file, \
+	$(wildcard models/examples/**/SendRequest-FhirMessageSigned.json), \
+	cat $(file) \
+	| jq 'del(.entry[] | select(.resource.resourceType == "Provenance"))' \
+	| jq 'del(.entry[0].resource.focus[0])' \
+	> ""`echo $(file) | sed "s/SendRequest-FhirMessageSigned/PrepareRequest-FhirMessageUnsigned/"`"";)
```
Comment (booshibooshi) [inline Makefile:62]: Stripped this back as we don't need yaml and json versions of all of these. We're using the existing symlinks to reference models directly.

Sync PrepareRequests from SendRequests, these are still validated by the fhir validator in the repo

File: Makefile [hunk 4]
```diff
@@ -80,7 +81,6 @@ build-coordinator:
 	cp coordinator/package.json coordinator/dist/
 	mkdir -p coordinator/dist/resources
 	cp coordinator/src/resources/ebxml_request.mustache coordinator/dist/resources/
-	poetry run scripts/update_coordinator_tests.py
```
Comment (booshibooshi) [inline Makefile:83]: Postman tests deprecated, so removed script to update these

File: coordinator/src/model/hl7-v3-datatypes-codes.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import {ElementCompact} from "xml-js";
-import {Text} from "./hl7-v3-datatypes-core";
+import {ElementCompact} from "xml-js"
```
Comment (booshibooshi) [inline coordinator/src/model/hl7-v3-datatypes-codes.ts:1]: semi-colon auto "fix" from linter

File: models/examples/example-1-repeat-dispensing/ConvertResponse-Hl7V3Message.xml [hunk 6]
```diff
@@ -43,7 +43,7 @@
         <id root="F68CEA42-E8A0-D799-C874-163048F0D390"/>
         <code code="163501000000109" codeSystem="2.16.840.1.113883.2.1.3.2.4.15" displayName="Prescription - FocusActOrEvent (record artifact)"/>
         <effectiveTime value="20200610120600"/>
-        <typeId extension="PORX_MT132004UK31" root="2.16.840.1.113883.2.1.3.2.4.18.7"/>
+        <typeId extension="PORX_MT132004SM31" root="2.16.840.1.113883.2.1.3.2.4.18.7"/>
```
Comment (booshibooshi) [inline models/examples/example-1-repeat-dispensing/ConvertResponse-Hl7V3Message.xml:46]: Have these been updated correctly?

File: azure-pipelines.yml [hunk 7]
```diff
@@ -22,15 +22,15 @@ pr:
...
+    timeoutInMinutes: "10"
     pool:
       vmImage: 'ubuntu-latest'
     variables:
       PIP_CACHE_DIR: ".poetry"
     steps:
       - checkout: self
         path: s
-        fetchDepth: 0
+        fetchDepth: "0"
```
Comment (ri-ch) [inline azure-pipelines.yml:33]: Do these not work as numbers?

File: coordinator/src/model/hl7-v3-prescriptions.ts [hunk 8]
```diff
@@ -167,7 +167,7 @@ export class ParentPrescription implements ElementCompact {
         this.id = id
         this.code = new codes.SnomedCode("163501000000109", "Prescription - FocusActOrEvent (record artifact)")
         this.effectiveTime = effectiveTime
-        this.typeId = new codes.TypeIdentifier("PORX_MT132004UK31")
+        this.typeId = new codes.TypeIdentifier("PORX_MT132004SM31")
```
Comment (ri-ch) [inline coordinator/src/model/hl7-v3-prescriptions.ts:170]: I think this change needs to come out, it was a mistake in my branch

File: models/examples/example-1-repeat-dispensing/ConvertResponse-Hl7V3Message.xml (hunk 6 above)
Comment (ri-ch) [reply models/examples/example-1-repeat-dispensing/ConvertResponse-Hl7V3Message.xml:46]: These are supposed to be the UK31 IDs, i double checked with the spine devs

File: tests/e2e/pact/specs/eps-sandbox.spec.ts [hunk 9]
```diff
@@ -17,20 +17,20 @@ jestpact.pactWith(
   /* eslint-disable  @typescript-eslint/no-explicit-any */
```
Comment (ri-ch) [inline tests/e2e/pact/specs/eps-sandbox.spec.ts:17]: Where is the explicit any?

Comment (booshibooshi) [reply tests/e2e/pact/specs/eps-sandbox.spec.ts:17]: the provider argument below is any, and this is part of the jest-pact library

File: azure-pipelines.yml (hunk 7 above)
Comment (booshibooshi) [reply azure-pipelines.yml:33]: VSCode complains at me, I think I have a linter for azure pipelines installed. Can switch them back to numbers if preferred, works for either.

Comment (ri-ch) [reply azure-pipelines.yml:33]: If it still works as strings, i have no strong feelings either way. Are consistent for the whole file?

File: coordinator/.eslintrc [hunk 10]
```diff
@@ -9,7 +12,7 @@
         "plugin:@typescript-eslint/eslint-recommended",
         "plugin:@typescript-eslint/recommended"
     ],
-    "env": {
-        "node": true
+    "rules": {
+        "semi": ["error", "never"]
```
Comment (ri-ch) [inline coordinator/.eslintrc:16]: Finally

**Pull request merged**